File: packages/square-mobile-payments/ios/Plugin/Classes/Results/IsAuthorizedResult.swift
import Foundation
import Capacitor

@objc public class IsAuthorizedResult: NSObject, Result {
    let authorized: Bool

    init(authorized: Bool) {
        self.authorized = authorized
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["authorized"] = authorized
        return result as AnyObject
    }
}

File: packages/square-mobile-payments/android/src/main/java/io/capawesome/capacitorjs/plugins/squaremobilepayments/classes/results/IsAuthorizedResult.java
package io.capawesome.capacitorjs.plugins.squaremobilepayments.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.squaremobilepayments.interfaces.Result;

public class IsAuthorizedResult implements Result {

    private final boolean authorized;

    public IsAuthorizedResult(boolean authorized) {
        this.authorized = authorized;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("authorized", authorized);
        return result;
    }
}
